rst_guide/advanced_syntax.rst

```diff
@@ -0,0 +1,1090 @@
+      :raw-html:`<span class="n">&nbsp;   c</span> <span class="o">=</span> <span class="n">a</span><span class="o">.</span><span class="n">copy</span><span class="p">()</span>   <span class="c1"># make a copy of a </span>`
+      :raw-html:`<span class="n">&nbsp;   c</span><span class="o">.</span><span class="n">update</span><span class="p">(</span><span class="n">b</span><span class="p">)</span>    <span class="c1"># modify keys and values of a with the ones from b</span>`
+      :raw-html:`<span class="k">&nbsp;   return</span> <span class="n">c</span>`
+      :raw-html:`&nbsp;`
+      :raw-html:`<span class="n">a</span> <span class="o">=</span> <span class="p">{</span> <span class="s1">'x'</span><span class="p">:</span> <span class="mi">1</span><span class="p">,</span> <span class="s1">'y'</span><span class="p">:</span> <span class="mi">2</span><span class="p">}</span>`
+      :raw-html:`<span class="n">b</span> <span class="o">=</span> <span class="p">{</span> <span class="s1">'y'</span><span class="p">:</span> <span class="mi">3</span><span class="p">,</span> <span class="s1">'z'</span><span class="p">:</span> <span class="mi">4</span><span class="p">}</span>`
+      :raw-html:`<span class="k">print</span><span class="p">(</span><span class="n">merge_two_dicts</span><span class="p">(</span><span class="n">a</span><span class="p">,</span> <span class="n">b</span><span class="p">))</span> <span class="c1"># {'y': 3, 'x': 1, 'z': 4}</span>`
+      :raw-html:`<span class="c1"># Appended comment</span>`
+      :raw-html:`</pre></div></td></tr></tbody></table></div></div>`
+
+.. rst-class:: pull-right
+
+| :glyphicon-info-sign:`\ ` **Read more about**  `Literal includes <https://www.sphinx-doc.org/en/1.6/markup/code.html#includes>`_.
+
+|
+
+Inline elements
+---------------
+.. _cross-reference:
+
+Cross-reference
+...............
+
+Hyperlinks can be used to link external websites or to link to different part of the course. However, for linking to different sections or chapters of the course, we use the ``:ref:`` and the ``doc`` roles. These two roles create a ling to the determined target indicated by the role. However, in order to make these cross-references work we should create or have specific targets.
+
+In the case of the ``ref:`` role, we define the target as ``.. _target-name:``. These syntax add an id to the element that is immediately below and create the target for the link.
+
+.. warning::
+  * You may add the targets above titles.
+  * The `name property <https://docutils.sourceforge.io/docs/ref/rst/directives.html#name>`_ add an id to the respective element and can be use as a target.
+
+In the case of the ``:doc:`` role the targets are the documents themselves. Therefore, the link to those documents should be an absolute or relative path.
+
+Cross-referencing :important:`syntax` consists of the ``:ref:`` or ``:doc:`` role, with the target as the interpreted text of the roles.
```

Comment:
I was wondering if there should be a code example for the `ref` and `doc` roles.